opm/autodiff/GeoProps.hpp

```diff
@@ -391,21 +383,29 @@ namespace Opm
             const double cellVolume = eclgrid->getCellVolume(cartesianCellIdx);
             ntg[cartesianCellIdx] *= cellVolume;
             double totalCellVolume = cellVolume;
+            double cellPoreVolume = porvData[cartesianCellIdx];
 
             // Average properties as long as there exist cells above
             // that has pore volume less than the MINPV threshold
             int cartesianCellIdxAbove = cartesianCellIdx - nx*ny;
             while ( cartesianCellIdxAbove >= 0 &&
-                 actnum[cartesianCellIdxAbove] > 0 &&
-                 porv[cartesianCellIdxAbove] < eclgrid->getMinpvValue() ) {
+                    porvData[cartesianCellIdxAbove] < eclgrid->getMinpvValue() ) {
 
-                // Volume weighted arithmetic average of NTG
                 const double cellAboveVolume = eclgrid->getCellVolume(cartesianCellIdxAbove);
+                if (actnumData[cartesianCellIdxAbove] == 0 && cellAboveVolume > 1e-3)
```

Comment:
This could break in the theoretical case of someone running on a faulted lab-scale or even SBED-scale grid, I'd therefore prefer a much smaller limit here to avoid the risk of practical consequences.